### PR TITLE
feat: some pyarrow dataframe methods

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #372 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

Introduces the following methods for pyarrow dataframe:

- collect
- drop
- drop_nulls
- lazy
- sort
- to_pandas

### CODE REVIEW

**Comment by FBruzzesi** [comment]
@MarcoGorelli CI fails due to `pandas<v1.0.0`.
In general I believe cases like this one can happen. One way of dealing with interoperability constraint could be to decorate pyarrow `to_pandas` method with something similar to [scikit-learn available_if](https://scikit-learn.org/stable/modules/generated/sklearn.utils.metaestimators.available_if.html) and support it only for certain versions installed?! WDYT?

**Comment by MarcoGorelli** [comment]
this makes me so happy!

**Comment by MarcoGorelli** [comment]
> One way of dealing with interoperability constraint could be to decorate pyarrow to_pandas method with something similar to [scikit-learn available_if](https://scikit-learn.org/stable/modules/generated/sklearn.utils.metaestimators.available_if.html) and support it only for certain versions installed?! WDYT?

Nice, I didn't know about that. sounds good! or just xfail some tests for now

File: narwhals/_arrow/dataframe.py [hunk 1]
```diff
@@ -112,3 +117,37 @@ def select(
         pa = get_pyarrow()
         df = pa.Table.from_arrays([s._series for s in new_series], names=names)
         return self._from_dataframe(df)
+
+    def drop(self, *columns: str | Iterable[str]) -> Self:
+        return self._from_dataframe(self._dataframe.drop(list(flatten(columns))))
```
Comment (FBruzzesi) [inline narwhals/_arrow/dataframe.py:122]: pyarrow `Table.drop` has been renamed to `drop_columns` from version 12.0.0, however for now `drop`  is kept for backwards compatibility.

**Review by MarcoGorelli — APPROVED** [review]
thanks so much, being able to run pyarrow with polars syntax would be 🔥 I think/hope this'll attract a few people to the project

**Pull request merged**